pandas/io/parsers.py

```diff
@@ -3400,7 +3570,7 @@ def _isindex(colspec):
                     colspec = orig_names[colspec]
                 if _isindex(colspec):
                     continue
-                data_dict[colspec] = converter(data_dict[colspec])
+                data_dict[colspec] = converter(np.array(data_dict[colspec]))
```

Comment:
use np.asarray

Reply:
why do you need this?

Reply:
converter doesn't work on pandas series, if i can remember correctly. doesn't hurt perf with non pyarrow case tho